File: python/src/plugins/erc20/radius_plugins/erc20/token.py
from typing import Dict, List, TypedDict


class ChainData(TypedDict):
    contractAddress: str


class Token(TypedDict):
    decimals: int
    symbol: str
    name: str
    chains: Dict[int, ChainData]


class ChainSpecificToken(TypedDict):
    chain_id: int
    decimals: int
    symbol: str
    name: str
    contract_address: str

USDC: Token = {
    "decimals": 6,
    "symbol": "USDC",
    "name": "USDC",
    "chains": {
        # TODO: Add USDC contract or utilize existing "RadToken" ERC20 contract
        1223953: {"contractAddress": "<TODO_ADD_RADIUS_USDC_CONTRACT_ADDRESS>"}
    },
}


def get_tokens_for_network(
    chain_id: int, tokens: List[Token]
) -> List[ChainSpecificToken]:
    result: List[ChainSpecificToken] = []

    for token in tokens:
        chain_data = token["chains"].get(chain_id)
        if chain_data:
            result.append(
                {
                    "chain_id": chain_id,
                    "decimals": token["decimals"],
                    "symbol": token["symbol"],
                    "name": token["name"],
                    "contract_address": chain_data["contractAddress"],
                }
            )

    return result

File: typescript/packages/plugins/erc20/src/token.ts
export type Token = {
  decimals: number;
  symbol: string;
  name: string;
  chains: Record<number, { contractAddress: `0x${string}` }>;
};

export type ChainSpecificToken = {
  chainId: number;
  decimals: number;
  symbol: string;
  name: string;
  contractAddress: `0x${string}`;
};

export const USDC: Token = {
  decimals: 6,
  symbol: 'USDC',
  name: 'USDC',
  chains: {
    '1223953': {
      // TODO: Fix this contract address
      contractAddress: '0x833589fCD6eDb6E08f4c7C32D4f71b54bdA02913',
    },
  },
};

export const RADUSD: Token = {
  decimals: 18,
  symbol: 'RADUSD',
  name: 'Radius Token',
  chains: {
    '1223953': {
      contractAddress: '0x9aeEa4f3025940dBdbf6863C7e16a23Ea95272a4',
    },
  },
};

export function getTokensForNetwork(chainId: number, tokens: Token[]): ChainSpecificToken[] {
  const result: ChainSpecificToken[] = [];

  for (const token of tokens) {
    const chainData = token.chains[chainId];
    if (chainData) {
      result.push({
        chainId: chainId,
        decimals: token.decimals,
        symbol: token.symbol,
        name: token.name,
        contractAddress: chainData.contractAddress,
      });
    }
  }

  return result;
}
